### PR TITLE
fix: make pinojs work pm2 in verdaccio

### PR DESCRIPTION
**Pull request by amitgilad3**
There is an open issue with pm2 and pinojs that logs are hijacked by pm2 if  we set pino.destination. 
In order to fix this according to https://github.com/pinojs/pino/issues/491, we  don't need to set destination.
so in case verdaccio logs are set to stdout pino destination  is not set

### CODE REVIEW

File: src/lib/logger/logger.ts [hunk 1]
```diff
@@ -1,8 +1,7 @@
 import pino from 'pino';
+import SonicBoom from 'sonic-boom';
```
Comment (juanpicado) [inline src/lib/logger/logger.ts:2]: This should be on the `package.json`, pnpm is quite strict in terms of dependencies are not estrictly defined.
```
└─ pino@npm:6.11.2
   └─ sonic-boom@npm:1.4.0 (via npm:^1.0.2)
```

**Review by juanpicado — CHANGES_REQUESTED** [review]
Unfortunately this solution does not work,

with `  - { type: file, path: verdaccio.log, level: http}` setup as logger, the `stdout` is enabled by default.

In Verdaccio 5 the logger is initialized twice (I've already fixed that in master), so Pino.js takes the default `stdout` by default and ignore the second setup call. You can check running this.

```
DEBUG=verdaccio:logger node debug/bootstrap.js
  verdaccio:logger setup logger +0ms
  verdaccio:logger logging stdout enabled +1ms
  verdaccio:logger has prettifier? true +0ms
 warn --- config file  - /home/user/.config/verdaccio/config.yaml
  verdaccio:logger setup logger +569ms
  verdaccio:logger logging file enabled +1ms
  verdaccio:logger has prettifier? true +0ms
(node:94779) Warning: deprecate: multiple logger configuration is deprecated, please check the migration guide.
(Use `node --trace-warnings ...` to show where the warning was created)
 warn --- Plugin successfully loaded: verdaccio-htpasswd
 warn --- Plugin successfully loaded: verdaccio-audit
 warn --- http address - http://localhost:4873/ - verdaccio/5.0.4
```

The log is redirected to stdout and the file is empty. 

Maybe make sense to also initialize setup only once very after the configuration has been parsed like I did here https://github.com/verdaccio/verdaccio/commit/19d272d10ec771665391c3d17c6fbe2812c27ecc

File: src/lib/logger/logger.ts (hunk 1 above)
Comment (juanpicado) [reply src/lib/logger/logger.ts:2]: Ignore this, we should not add `sonic-boom` dependency at all or use it as import, if Pino.js change that, then breaking change would be hard to fix.

**Comment by amitgilad3** [comment]
Thanks @juanpicado for the update, I'll look what you did and try to fix it accordingly

**Comment by juanpicado** [comment]
closing this, logger not longer in 5.x branch

**Pull request closed without merging**